tests/pthread/test_pthread_atexit.c

```diff
@@ -0,0 +1,42 @@
+#include <assert.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "emscripten.h"
+#include "emscripten/threading.h"
+
+static _Atomic int threadCounter = 0;
+static _Atomic int running = 1;
+static pthread_t thread;
+
+void *workerThread(void* arg) {
+  threadCounter++;
+
+  while (running)
+    emscripten_thread_sleep(1000);
```

Comment:
Why 1000 here?  If we want to test to run as fast as possible should we use a shorter value?  Or maybe pthread_cond_wait instead?

Reply:
Originally this test case was based on the example from https://www.luke1410.de/blog/2017/02/the-trouble-of-separate-module-atexit-stacks/. But indeed, this can be adjusted by using `pthread_cond_wait` instead, which is much neater.

Done with commit 786473b83591d39792dcb197bc8add00692d57a7.